core/src/Kokkos_Half.hpp

```diff
@@ -61,8 +61,10 @@
 #define KOKKOS_IMPL_HALF_TYPE_DEFINED
 namespace Kokkos {
 namespace Experimental {
-using half_device_type             = float;
-using half_t                       = half_device_type;
```

Comment:
This is an interesting thought, but I think we probably shouldn't do that. If we actually need the impl_type generically we may want to add a trait typedef somewhere. But if its just that one test we can work around there.